lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -310,6 +313,71 @@ sub settings {
     $self->render('test/settings');
 }
 
+=over 4
+
+=item show_filesrc()
+
+Returns the context of a config file of the selected job.
+So this works in the same way as the test module source.
+
+=back
+
+=cut
+
+sub show_filesrc {
+    my ($self)      = @_;
+    my $job         = $self->_stash_job or return $self->reply->not_found;
+    my $jobid       = $self->param('testid');
+    my $dir         = $self->stash('dir');
+    my $data_uri    = $self->stash('link_path');
+    my $testcasedir = testcasedir($job->DISTRI, $job->VERSION);
+    if (my $casedir = $job->settings->single({key => 'CASEDIR'})) {
+        my $casedir_url = Mojo::URL->new($casedir->value);
+        # if CASEDIR points to a remote location let's assume it is a git repo
+        # that we can reference like gitlab/github
+        last unless $casedir_url->scheme;
+        my $refspec = $casedir_url->fragment;
+        # try to read vars.json from resultdir and replace branch by actual git hash if possible
+        eval {
+            my $vars_json = Mojo::File->new($job->result_dir(), 'vars.json')->slurp;
+            my $vars      = decode_json($vars_json);
+            $refspec = $vars->{TEST_GIT_HASH};
+        };
+        my $src_path;
+        # Use the testcasedir to determine the correct path
+        if (-d path($testcasedir)->child($dir)) {
+            $src_path = path('/blob', $refspec, $dir, $data_uri);
+        }
+        else {
+            $src_path
+              = path('/blob', $refspec, $self->app->config->{job_settings_ui}->{default_data_dir}, $dir, $data_uri);
+        }
+        # github treats '.git' as optional extension which needs to be stripped
+        $casedir_url->path($casedir_url->path =~ s/\.git//r . $src_path);
+        $casedir_url->fragment('');
+        return $self->redirect_to($casedir_url);
+    }
+
+    my $setting_file_path;
+
+    if (-d path($testcasedir)->child($dir)) {
+        $setting_file_path = path($testcasedir, $dir, $data_uri);
+    }
+    else {
+        $setting_file_path
+          = path($testcasedir, $self->app->config->{job_settings_ui}->{default_data_dir}, $dir, $data_uri);
```

Comment:
It would also be nice to avoid repeating `$self->app->config->{job_settings_ui}->{default_data_dir}, $dir, $data_uri`.

Reply:
will do